arm-hdinsight/2015-03-01-preview/swagger/cluster.json

```diff
@@ -0,0 +1,1187 @@
+          "description": "Gets or sets the port to connect to."
+        }
+      }
+    },
+    "ClusterGetProperties": {
+      "properties": {
+        "clusterVersion": {
+          "type": "string",
+          "description": "Gets or sets the version of the cluster."
+        },
+        "osType": {
+          "type": "string",
+          "description": "Gets or sets the type of operating system.",
+          "enum": [
+            "Windows",
+            "Linux"
+          ],
+          "x-ms-enum": {
+            "name": "OSType",
+            "modelAsString": false
+          }
+        },
+        "tier": {
+          "type": "string",
+          "description": "Gets or sets the cluster tier.",
+          "enum": [
+            "Standard",
+            "Premium"
+          ],
+          "x-ms-enum": {
+            "name": "Tier",
+            "modelAsString": false
+          }
+        },
+        "clusterDefinition": {
+          "$ref": "#/definitions/ClusterDefinition",
+          "description": "Gets or sets the cluster definition."
+        },
+        "securityProfile": {
+          "$ref": "#/definitions/SecurityProfile",
+          "description": "Gets or sets the security profile."
+        },
+        "computeProfile": {
+          "$ref": "#/definitions/ComputeProfile",
+          "description": "Gets or sets the compute profile."
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "Gets or sets the provisioning state, which only appears in the response.",
+          "enum": [
+            "InProgress",
+            "Failed",
+            "Succeeded",
+            "Canceled",
+            "Deleting"
+          ],
+          "x-ms-enum": {
+            "name": "HDInsightClusterProvisioningState",
+            "modelAsString": false
+          }
+        },
+        "createdDate": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets or sets the date on which the cluster was created."
+        },
+        "clusterState": {
+          "type": "string",
+          "description": "Gets or sets the state of the cluster."
+        },
+        "quotaInfo": {
+          "$ref": "#/definitions/QuotaInfo",
+          "description": "Gets or sets the quota information."
+        },
+        "errors": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ErrorInfo"
+          },
+          "description": "Gets or sets the list of errors."
+        },
+        "connectivityEndpoints": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConnectivityEndpoint"
+          },
+          "description": "Gets or sets the list of connectivity endpoints."
+        }
+      },
+      "required": [
+        "clusterDefinition"
+      ],
+      "description": "Describes the properties of a cluster."
+    },
+    "Cluster": {
+      "properties": {
+        "id": {

```

Comment:
This looks like a "Resource" please create a "Resource" model and make an allOf on it here